pkg/sql/plan/function/baseTemplate.go

```diff
@@ -1609,6 +1609,54 @@ func opBinaryBytesBytesToFixedWithErrorCheck[Tr types.FixedSizeTExceptStrType](
 	return nil
 }
 
+func compareVarlenaEqual(parameters []*vector.Vector, result vector.FunctionResultWrapper, _ *process.Process, length int) error {
+	rs := vector.MustFunctionResult[bool](result)
+	rsVec := rs.GetResultVector()
+	rss := vector.MustFixedCol[bool](rsVec)
+
+	v1, v2 := parameters[0], parameters[1]
+	if v1.IsConstNull() || v2.IsConstNull() {
+		nulls.AddRange(rsVec.GetNulls(), 0, uint64(length))
+		return nil
+	}
+	col1, _ := vector.MustVarlenaRawData(v1)
+	col2, _ := vector.MustVarlenaRawData(v2)
+
+	if v1.IsConst() && v2.IsConst() {
+		p1 := col1[0].UnsafePtr()
+		p2 := col2[0].UnsafePtr()
+		ret := *(*[3]int64)(p1) == *(*[3]int64)(p2)
+		for i := uint64(0); i < uint64(length); i++ {
+			rss[i] = ret
+		}
+		return nil
+	}
+
+	if !v1.IsConst() && !v2.IsConst() {
+		for i := 0; i < length; i++ {
+			p1 := col1[i].UnsafePtr()
+			p2 := col2[i].UnsafePtr()
+			rss[i] = *(*[3]int64)(p1) == *(*[3]int64)(p2)
+		}
+		nulls.Or(parameters[0].GetNulls(), parameters[1].GetNulls(), rsVec.GetNulls())
+	} else if v1.IsConst() {
+		p1 := col1[0].UnsafePtr()
+		for i := 0; i < length; i++ {
+			p2 := col2[i].UnsafePtr()
+			rss[i] = *(*[3]int64)(p1) == *(*[3]int64)(p2)
+		}
+		nulls.Or(rsVec.GetNulls(), parameters[1].GetNulls(), rsVec.GetNulls())
+	} else {
+		p2 := col2[0].UnsafePtr()
```

Comment:
Save here.    This loop if you have to go into vector details to get max perf, the following code belongs to the vector package.

Reply:
fixed